src/lightning/pytorch/trainer/trainer.py

```diff
@@ -112,13 +112,13 @@ def __init__(
         limit_test_batches: Optional[Union[int, float]] = None,
         limit_predict_batches: Optional[Union[int, float]] = None,
         val_check_interval: Optional[Union[int, float]] = None,
-        log_every_n_steps: int = 50,
+        log_every_n_steps: Optional[int] = None,
```

Comment:
Why can't we leave these defaults? This makes everything less readable, a step back again from the goals of 2.0 in my opinion. I don't think we need to do a special validation below for every flag. IMO, if `barebones=True`, we show the warning/info message that all other settings get bypassed, regardless of their setting. The same principle applied to fast_dev_run previously.

Reply:
To avoid users mistakenly thinking that `Trainer(logger=TensorBoardLogger(), barebones=True)` works.

I agree it's less readable, but the current implementation makes it more fool-proof.

If we did a `BarebonesTrainer`, it wouldn't expose these arguments, and the Trainer argument defaults would stay as they are. But that's a different solution

Reply:
Barebones is False by default, so a user has to activate it manually for it to run. So they learn what it does by reading the docstring and then through the info message that shows up. So it's not like someone would blindly enable a feature they don't know what it does, and then be surprised that tensorboard shows no logs.

Reply:
I disagree. Several times people have left trainer flags enabled and then complained about issues related to them, without noticing that the flag is active.

Maybe it was an accident, maybe it's a new user who copy-pasted an example, maybe an issue in your CLI configuration, ... or maybe they just don't really understand exactly what it does!

Reply:
Yeah not ideal but I agree it's the safest. If we could inspect what gets passed into the constructor that would be great but there doesn't seem to be a good way.